src/core/model.jl

```diff
@@ -190,13 +194,73 @@ Return an iterator over all agents of the model.
 """
 allagents(model) = values(model.agents)
 
-
 """
     allids(model)
 Return an iterator over all agent IDs of the model.
 """
 allids(model) = keys(model.agents)
 
+#######################################################################################
+# %% Higher order collections
+#######################################################################################
+export group_agent_iter, group_id_iter, group_agent_map, group_id_map
+
+"""
+    group_id_iter(order::Int, model::ABM)
+
+Return an iterator over all agent IDs of the model, group by order. When `order = 2`, the
+iterator returns agent pairs, and when `order = 3`: agent triples.
+
+See also [`group_agent_iter`](@ref).
+"""
+group_id_iter(order::Int, model::ABM) =
+    Iterators.product((by_id(model) for _ in 1:order)...)
+
+"""
+    group_agent_iter(order::Int, model::ABM)
+
+Return an iterator over all agents of the model, group by order. When `order = 2`, the
+iterator returns ID pairs, e.g `(1, 2)` and when `order = 3`: ID triples, e.g.
+`(1, 7, 8)`.
+
+See also [`group_id_iter`](@ref).
+"""
```

Comment:
The example given is not what this function does, but what the function `group_id_iter` does.

Reply:
We should move the example to the `group_id_iter` and in this docstring only cross-ref` group_id_iter` and say "this function does the same but with agents. This is how we deal with `nearby_ids` and `nearby_agents`.